frontend/lib/features/send/send_page.dart

```diff
@@ -33,22 +36,43 @@ class SendPage extends HookWidget {
               padding: const EdgeInsets.symmetric(vertical: Grid.xs),
               child: NumberPad(
                 onKeyPressed: (key) {
-                  sendAmount.value = (sendAmount.value == '\$0')
-                      ? '\$$key'
-                      : '${sendAmount.value}$key';
+                  isValidKeyPress.value = true;
+                  isValidKeyPress.value =
+                      NumberPadInputValidationUtil.validateKeyPress(
+                          sendAmount.value, key);
+
+                  if (isValidKeyPress.value) {
+                    sendAmount.value = (sendAmount.value == '0')
+                        ? key == '.'
+                            ? key.padLeft(2, '0')
+                            : key
+                        : sendAmount.value + key;
+                  }
                 },
                 onDeletePressed: () {
-                  sendAmount.value = (sendAmount.value.length > 2)
-                      ? sendAmount.value
-                          .substring(0, sendAmount.value.length - 1)
-                      : '\$0';
+                  isValidKeyPress.value = true;
+                  isValidKeyPress.value =
+                      NumberPadInputValidationUtil.validateDeletePress(
+                          sendAmount.value);
+
+                  if (isValidKeyPress.value) {
+                    sendAmount.value = (sendAmount.value.length > 1)
+                        ? sendAmount.value
+                            .substring(0, sendAmount.value.length - 1)
+                        : '0';
+                  }
```

Comment:
@kirahsapong what's this bit doing? It seems like it sets `isValidKeyPress.value = true` then immediately sets it again to the result of `validateDeletePress`

Reply:
It feels like the `InputNumberPadInputAnimation` should be responsible for it's own resetting as needed. So the outside world would just need to set when it's invalid and when it's not. Not that we need to change anything at this point.

Also these animation type things can be a bit of a time suck so it might be good to get more of the core functionality in, then circle back to the polish and animation stuff in the future.

Reply:
OK cool We can dig into this later too. It feels odd to me that `send_page` is responsible for resetting the state of the `InputNumberPadInputAnimation` if I'm reading this correctly. But we can refactor later.